device/nokia/x86_64-nokia_ixr7250e_sup-r0/platform_reboot

```diff
@@ -1,15 +1,39 @@
 #!/bin/bash
-echo "Rebooting all Linecards"
-python3 -c 'import sonic_platform.platform; platform_chassis = sonic_platform.platform.Platform().get_chassis(); platform_chassis.reboot_imms()'
-sleep 3
+
+DEVICE_MGR_REBOOT_FILE="/tmp/device_mgr_reboot"
+
+update_reboot_cause()
+{
+    DEVICE_MGR_REBOOT_FILE=/tmp/device_mgr_reboot
+    REBOOT_CAUSE_FILE=/host/reboot-cause/reboot-cause.txt
+    DEVICE_REBOOT_CAUSE_FILE=/etc/opt/srlinux/reboot-cause.txt
+    if [ -e  $DEVICE_MGR_REBOOT_FILE ]; then
+        if [ -e $DEVICE_REBOOT_CAUSE_FILE ]; then
+            cp -f $DEVICE_REBOOT_CAUSE_FILE $REBOOT_CAUSE_FILE
+        fi
+        rm -f $DEVICE_MGR_REBOOT_FILE
+    else
+        touch /etc/opt/srlinux/devmgr_reboot_cause.done
+        rm -f $DEVICE_REBOOT_CAUSE_FILE &> /dev/null
+    fi
+    sync
+}
+
+if [ ! -e $DEVICE_MGR_REBOOT_FILE ]; then
```

Comment:
Why checking for this file DEVICE_MGR_REBOOT_FILE ? Is this to make sure we reboot linecards only when user issues a reboot command ?

Reply:
This is used to differentiate the "sudo reboot" command issued by user entering "sudo reboot"   or it is issued by NKD which is triggered by PMON API.  This is to avoid the redundant rebooting linecard.